fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/client/item/v1/CustomModeledArmorRegistry.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.item.v1;
+
+import java.util.Arrays;
+
+import net.minecraft.client.render.entity.model.BipedEntityModel;
+import net.minecraft.entity.EquipmentSlot;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.fabricmc.fabric.impl.client.item.CustomModeledArmorRegistryImpl;
+
+@Environment(EnvType.CLIENT)
+public final class CustomModeledArmorRegistry {
+	private CustomModeledArmorRegistry() {
+	}
+
+	/**
+	 * Registers a provider for custom armor models for an item.
+	 *
+	 * @param provider the provider for the model
+	 * @param items    the items to be registered for
+	 */
+	public static void registerCustomArmorModel(/* @Nullable */ Provider provider, Item... items) {
```

Comment:
I think we should use `ItemConvertible` here in case someone decides to make a literal armor block so you don't need an `asItem` call.

Reply:
I think you should null check the items varargs.

Reply:
True

Reply:
the null check can implicitly happen when iterating